Src/Base/AMReX_FabArrayBase.cpp

```diff
@@ -765,7 +764,7 @@ FabArrayBase::FB::define_fb(const FabArrayBase& fa)
             // with boxes for overlapping ghost nodes.
             Box ba_ksnd = ba[ksnd];
             ba_ksnd.grow(1);
-            const Box dst_bx_ng = (ba_ksnd & (bxrcv_ng + (*pit))) - (*pit);
+            const Box dst_bx_ng = (ba_ksnd & (bxrcv + (*pit))) - (*pit);
 		    const BoxList& bl_ng = amrex::boxDiff(dst_bx_ng, vbx_ng);
             bl.join(ba.complementIn(bl_ng));
```

Comment:
```suggestion
            const BoxList& bltmp = ba.complementIn(dst_bx_ng);
            for (auto const& btmp : bltmp) {
                bl.join(amrex::boxDiff(btmp,vbx_ng));
           }
```
I think this is better for performance.  The reason is that `BoxArray::complemenIn` is expensive, so we want to avoid doing it on a BoxList.  We can then also remove the new `BoxArray::complementIn(BoxList const&)` function.  If it's there, people might use it without realizing the performance issue.

Reply:
Ok, makes sense. I've updated the change here and for the parallel part.